### PR TITLE
Projections

### PR DESCRIPTION
**Pull request by hernando**

### CODE REVIEW

File: brion/blueConfig.h [hunk 1]
```diff
@@ -116,10 +116,17 @@ class BlueConfig : public boost::noncopyable
...
-     * @version 1.7
+     * @return the URI to the location of synapse nrn files for projections from
+     *         external circuits (e.g. thalamocortical projections)
+     * @version 2.0
      */
+    BRION_API URI getProjectionSource(const std::string& name) const;
+
+    /**
+      * @return the full path to the morphology database. A suffix may be
+      *         prepended to the to the bare path from the BlueConfig.
```
Comment (tribal-tec) [inline brion/blueConfig.h]: to the to the?

File: brain/circuit.h [hunk 2]
```diff
@@ -157,6 +157,27 @@ class Circuit
...
+     *        first operation that tries to access the data.
+     * \if pybind
+     * @param prefetch which synapse data to preload
+     * \else
+     * @param prefetch which synapse data to load on SynapsesStream.read()
+     * @return synapse data stream. The pre-synaptic GIDs in the synapses refer
+     *         to cells in the projecting circuit.
+     * \endif
+     */
+    BRAIN_API SynapsesStream getAfferentProjections(
```
Comment (tribal-tec) [inline brain/circuit.h]: we don't get a confusion with the existing getProjectedSynapses() in terms of similar names I hope.

File: brain/detail/synapsesStream.h [hunk 3]
```diff
@@ -69,10 +82,21 @@ struct SynapsesStream
...
-        });
+
+        if (_source.empty())
+        {
+            return std::async(std::launch::async, [&, start, end] {
+                return Synapses(_circuit, GIDSet(start, end), _filterGIDs,
+                                _afferent, _prefetch);
+            });
+        }
+        else
```
Comment (tribal-tec) [inline brain/detail/synapsesStream.h]: else not needed

File: brain/synapses.cpp [hunk 4]
```diff
@@ -123,14 +139,18 @@ struct Synapses::Impl : public Synapses::BaseImpl
             return;
 
         const brion::Synapse& synapseAttributes =
-            _circuit._impl->getSynapseAttributes(_afferent);
-        const brion::Synapse* synapseExtra = _circuit._impl->getSynapseExtra();
+            _source.empty()
```
Comment (tribal-tec) [inline brain/synapses.cpp]: rename _source to _externalSource or similar? It's not quite obvious deeply in the code what this means.

File: brain/circuit.h (hunk 2 above)
Comment (hernando) [inline brain/circuit.h]: Difficult to say, if you have a better suggestion it will be welcomed.

File: brain/synapses.cpp (hunk 4 above)
Comment (hernando) [inline brain/synapses.cpp]: I'll do

**Pull request merged**